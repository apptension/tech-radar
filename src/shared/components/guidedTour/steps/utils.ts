import { color } from '../../../../theme';
import { highlightBlip, unhighlightBlip } from '../../../utils/radarUtils';
import { RadarTechnology } from './../../radar/radar.types';

export const highlightBlips = (list: RadarTechnology[]) => {
  list.forEach((tech) => {
    highlightBlip({ id: tech?.id || '', ring: Number(tech?.ring), scale: false });
  });
};

export const unHighlightBlips = (list: RadarTechnology[]) => {
  list.forEach((tech) => {
    unhighlightBlip({
      id: tech?.id || '',
      ring: Number(tech?.ring),
      color: color.mineShaft,
    });
  });
};
