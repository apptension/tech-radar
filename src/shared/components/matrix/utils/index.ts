import { Skill } from '../types';

export const checkIfSkillIsAdded = (skills: Skill[], skillValue: string) =>
  skillValue !== skills.find((item) => item.value === skillValue)?.value;

export const findSkill = (skillValue: string, skills: Skill[]) => {
  return skills.find((skill) => skill.value === skillValue) as Skill;
};

export const mapSkillToSkillWithVisibility = (skill: Skill) => ({ ...skill, isVisible: true });

export const getUserSkillsFromIds = (skillIds: string[], skills: Skill[]) =>
  skillIds.map((skillId) => {
    const skill = findSkill(skillId, skills);
    return mapSkillToSkillWithVisibility(skill);
  });

export const mapSkillsToValues = (skills: Skill[]) => skills.map(({ value }) => value);
