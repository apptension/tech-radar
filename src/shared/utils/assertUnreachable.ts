export const assertUnreachable = (value: never, message: string): never => {
  throw new Error(message);
};
