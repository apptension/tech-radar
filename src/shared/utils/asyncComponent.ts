import loadable, { DefaultComponent } from '@loadable/component';
import { isNil } from 'ramda';

type DefaultExportModule = DefaultComponent<unknown>;
type NamedExportModule = Record<string, DefaultComponent<unknown>>;

type NamedExportLoader = () => Promise<NamedExportModule>;
type DefaultExportLoader = () => Promise<DefaultExportModule>;
type Module = NamedExportModule | DefaultExportModule;


export const asyncComponent = (asyncLoader: NamedExportLoader | DefaultExportLoader, exportedAs?: string) => {
  const isNamedExportModule = (module: Module, exportedAs?: string): module is NamedExportModule => {
    return !isNil(exportedAs);
  };

  return loadable(async () => {
    const module = await asyncLoader();
    return isNamedExportModule(module, exportedAs) && exportedAs ? module[exportedAs] : module as DefaultExportModule;
  });
};
