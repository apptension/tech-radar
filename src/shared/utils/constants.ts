export const transition = '200ms ease-in-out';
export const tooltipDelay = '0.5s';
