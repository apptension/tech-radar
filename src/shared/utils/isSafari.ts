export const isSafari = () => {
  // @ts-ignore
  return window.safari !== undefined;
};
