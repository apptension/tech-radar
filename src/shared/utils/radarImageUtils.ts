export const getMoveImageBy = (isMobile: boolean): number => {
  const MIN_WINDOW_HEIGHT_FOR_MOBILE_RADAR_IMAGE = 740;
  const windowImageHeightDifference = window.innerHeight - MIN_WINDOW_HEIGHT_FOR_MOBILE_RADAR_IMAGE;
  const isWindowTooShort = windowImageHeightDifference < 0;
  return isMobile && isWindowTooShort ? windowImageHeightDifference : 0;
};
