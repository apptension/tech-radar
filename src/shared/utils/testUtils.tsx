import React, { ReactNode, ReactElement } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { identity } from 'ramda';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import { Route, Router } from 'react-router';
import { IntlProvider } from 'react-intl';
import produce from 'immer';

import { DEFAULT_LOCALE, translationMessages, MessagesObject } from '../../i18n';
import { store as fixturesStore } from '../../mocks/store';
import createReducer, { GlobalState } from '../../config/reducers';

export const PLACEHOLDER_TEST_ID = 'content';
export const PLACEHOLDER_CONTENT = <span data-testid="content">content</span>;

export const spiedHistory = (route = '/') => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const pushSpy = jest.spyOn(history, 'push');
  return {
    history,
    pushSpy,
  };
};

interface ContextData {
  router?: {
    url?: string;
    routePath?: string;
    history?: MemoryHistory;
  };
  store?: GlobalState;
  messages?: MessagesObject;
}

interface ProvidersWrapperProps {
  children: ReactNode;
  context?: ContextData;
}

export const ProvidersWrapper = ({ children, context = {} }: ProvidersWrapperProps) => {
  const { router = {}, store = fixturesStore, messages } = context;
  const { url = `/${DEFAULT_LOCALE}`, routePath = '/:lang/', history } = router;

  const routerHistory: MemoryHistory = history ?? createMemoryHistory({ initialEntries: [url] });

  const intlProviderMockProps = {
    locale: DEFAULT_LOCALE,
    messages: messages ?? translationMessages[DEFAULT_LOCALE],
  };

  return (
    <Router history={routerHistory}>
      <HelmetProvider>
        <IntlProvider {...intlProviderMockProps}>
          <Provider store={createStore(createReducer(), produce(store, identity))}>
            <Route path={routePath}>{children}</Route>
          </Provider>
        </IntlProvider>
      </HelmetProvider>
    </Router>
  );
};

export const makeContextRenderer = <T, _>(component: (props: T | Record<string, never>) => ReactElement) => (
  props?: T,
  context?: ContextData
) =>
  render(component(props ?? {}), {
    wrapper: ({ children }) => <ProvidersWrapper context={context}>{children}</ProvidersWrapper>,
  });

export const makePropsRenderer = <T, _>(component: (props: T | Record<string, never>) => ReactElement) => (props?: T) =>
  render(component(props ?? {}));
