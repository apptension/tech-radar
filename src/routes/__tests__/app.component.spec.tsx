import React from 'react';
import { makeContextRenderer, PLACEHOLDER_CONTENT } from '../../shared/utils/testUtils';
import { AppComponent, AppComponentProps } from '../app.component';

const defaultProps: AppComponentProps = {
  children: PLACEHOLDER_CONTENT,
};

describe('App: Component', () => {
  const component = (props: Partial<AppComponentProps>) => <AppComponent {...defaultProps} {...props} />;
  const render = makeContextRenderer(component);

  it('should render App when language is set', () => {
    const { container } = render();
    expect(container).toMatchSnapshot();
  });
});
