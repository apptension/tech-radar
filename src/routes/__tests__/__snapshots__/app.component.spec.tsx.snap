// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Component should render App when language is set 1`] = `
<div>
  <span
    data-testid="content"
  >
    content
  </span>
</div>
`;
